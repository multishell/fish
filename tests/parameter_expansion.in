# basic expansion test
echo {}
echo {apple}
echo {apple,orange}

# expansion tests with spaces
echo {apple, orange}
echo { apple, orange, banana }

# expansion with spaces and cartesian products
echo \'{ hello , world }\'

# expansion with escapes
for phrase in {good\,,   beautiful ,morning}; echo -n "$phrase "; end | string trim;
for phrase in {goodbye\,,\ cruel\ ,world\n}; echo -n $phrase; end;

# whitespace within entries converted to spaces in a single entry
for foo in {a, hello
world }
	echo \'$foo\'
end

# dual expansion cartesian product
echo { alpha, beta }\ {lambda, gamma }, |  string replace -r ',$' ''

# expansion with subshells
for name in { (echo Meg), (echo Jo) }
	echo $name
end

# subshells with expansion
for name in (for name in {Beth, Amy}; printf "$name\n"; end); printf "$name\n"; end

echo {{a,b}}
# vim: set ft=fish:
