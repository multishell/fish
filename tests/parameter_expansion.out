{}
{apple}
apple orange
apple orange
apple orange banana
'hello' 'world'
good, beautiful morning
goodbye, cruel world
'a'
'hello world'
alpha lambda, beta lambda, alpha gamma, beta gamma
Meg
Jo
Beth
Amy
{a} {b}
